pkg/containerutil/cp_linux.go

```diff
@@ -247,3 +275,46 @@ func mountSnapshotForContainer(ctx context.Context, client *containerd.Client, c
 	}
 	return tempDir, cleanup, nil
 }
+
+func getContainerMountInfo(ctx context.Context, con containerd.Container, containerPath string) (string, string, error) {
+	filePath := filepath.Clean(containerPath)
+	spec, err := con.Spec(ctx)
+	if err != nil {
+		return "", "", err
+	}
+	if spec.Root.Readonly {
+		return "", "", fmt.Errorf("container rootfs: %s is marked read-only", spec.Root.Path)
+	}
+
+	for _, mount := range spec.Mounts {
+		if isParentDir(filePath, mount.Destination) {
+			for _, option := range mount.Options {
+				if option == "ro" {
+					return "", "", fmt.Errorf("mount point %s is marked read-only", filePath)
+				}
+			}
+			return mount.Source, mount.Destination, nil
+		}
+	}
+	return "", "", nil
+}
+
+func isParentDir(filePath, potentialParent string) bool {
+	filePath = filepath.Clean(filePath)
+	potentialParent = filepath.Clean(potentialParent)
+	if filePath == potentialParent {
+		return true
+	}
+	if filePath == "/" {
+		return false
+	}
+	return isParentDir(path.Dir(filePath), potentialParent)
+}
+
+// When the path is in volume remove directory that volume is mounted on from the path
+func handleVolumePaths(container2host bool, dst string, src string, destination string) (string, string) {
```

Comment:
`dst` and `destination` are really confusing, please choose more meaningful names